packages/material-ui-system/src/spacing.js

```diff
@@ -22,8 +22,28 @@ const directions = {
 // From 300,000 ops/sec
 // To 350,000 ops/sec
 const getCssProperties = memoize(prop => {
+  switch (prop) {
```

Comment:
Would using a map be more space and runtime efficient?
```jsx
const aliases = {
  marginX: 'mx',
  marginY:  'my',
  paddingX: 'px',
  paddingY: 'py',
},
```

Reply:
One of my favourite things is swapping switches for maps.

Reply:
I like doing that too but a map wouldn't work very well without also specifying the options (which I was trying to avoid) because I would have to do a check later for undefined if nothing matched.

also I decided not to do this because there's a comment a little later about not wanting to use a map (for something else, where (frankly) a `Set` would be best) because of fear of increasing the bundle size.  Not sure why or what that is really about but... all the same... I was trying to keep it as simple as possible.